src/engine/sparqlExpressions/LangExpression.h

```diff
@@ -1,39 +1,63 @@
-//  Copyright 2022, University of Freiburg,
+//  Copyright 2022-2024 University of Freiburg,
 //                  Chair of Algorithms and Data Structures.
-//  Author: Johannes Kalmbach <kalmbach@cs.uni-freiburg.de>
+//  Author: Johannes Kalmbach <kalmbach@cs.uni-freiburg.de> (2022)
+//  Author: Hannes Baumann <baumannh@informatik.uni-freiburg.de> (2024)
 
-// Test file for the LangExpression class: LanguageExpressionsTests.h
+// Unit tests for the functionality from this file can be
+// found in LanguageExpressionsTest.h
 
 #pragma once
 
-#include "engine/sparqlExpressions/SparqlExpression.h"
+#include "./engine/sparqlExpressions/LiteralExpression.h"
+#include "./engine/sparqlExpressions/NaryExpressionImpl.h"
+#include "./engine/sparqlExpressions/SparqlExpressionTypes.h"
 
 namespace sparqlExpression {
-
-class LangExpression : public SparqlExpression {
- private:
-  // The stored variable.
-  Variable variable_;
-  SparqlExpression::Ptr child_;
-
+namespace detail::langImpl {
```

Comment:
I was first thinking about only implementing this in the .cpp file, and making a `makeLangExpression`fucntion, but that doesn't quite work that simply, because we also need the
functionality of the added function in the parser module. 
It can be done , but is more involved, maybe add a TODO:)

Reply:
@joka921 so changing it to a full built-in function?

Reply:
No, the problem is,
That the parser needs the special functions of a `LangExpression` to determine the language filters etc. And because of this, we cannot completely hide the class and just expose a function, that returns `SparqlExpression::Ptr`.